packaging/hip-targets-release.cmake

```diff
@@ -30,15 +30,14 @@ list(APPEND _IMPORT_CHECK_FILES_FOR_hip::hip_hcc_static "${_IMPORT_PREFIX}/lib/l
 
 # Import target "hip::hip_hcc" for configuration "Release"
 set_property(TARGET hip::hip_hcc APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
-if(HIP_COMPILER STREQUAL "hcc" OR HIP_COMPILER STREQUAL "clang")
+if(CMAKE_CXX_COMPILER MATCHES "(.*)hcc" OR CMAKE_CXX_COMPILER MATCHES "(.*)clang(.*)")
```

Comment:
This should be

if (HIP_RUNTIME STREQUAL 'HCC')

whether the target requires these libs depend on whether the runtime is HIP/HCC, not depending on the compiler.

Reply:
@yxsamliu The problem I believe is that adding hcc::hccrt also implies -hc flag for compilation. However there are multiple math libraries which use HIP APIs in host code but compile using standard c/c++ compilers. So to handle this we need to only link libhip_hcc.so, but not add the -hc flag. We can only do this by testing which compiler is being used. However given that this is a shared file, we should guard this with a ```AND HIP_RUNTIME STREQUAL 'HCC'```

Reply:
If hcc::hccrt implies -hc flag, then only HCC compiler can handle it, since HIP-Clang does not support -hc flag.

However, don't math libs built with standard C/C++ compilers need to link to HCC runtime libraries when they link with HIP/HCC runtime library?